crates/spirv-std/src/runtime_array.rs

```diff
@@ -0,0 +1,39 @@
+use core::marker::PhantomData;
+
+#[spirv(runtime_array)]
+pub struct RuntimeArray<T> {
+    // spooky! this field does not exist, so if it's referenced in rust code, things will explode
+    _do_not_touch: u32,
+    _phantom: PhantomData<T>,
+}
```

Comment:
Calling this `RuntimeArray` is a bit confusing, someone was already in chat asking whether this was used for runtime data arrays. `RuntimeDescriptorArray` is a bit verbose, but it's also unambiguous, and for descriptor arrays of buffers this type needs to insert a block decorated struct around the buffer and have special indexing behaviour to handle that.

The `u32` field is a bit strange, descriptor arrays, like descriptors, don't really have size/alignment per se.  Since the compiler is handling this specially anyways, is it necessary?

Reply:
> Since the compiler is handling this specially anyways, is it necessary?

You have to avoid making it a ZST, because those can be assumed to be data-less tokens, i.e. they have only "existence" but no memory/register representation. For example, `(SomeZST, f32)` is represented as *just* an `f32`, and so there is no way to recover any data the backend may want to keep in the ZST type.

The other option (instead of a dummy field) is a `!Sized` type, via `extern { type Foo; }`, which we should maybe move to at some point.

Reply:
Yeah, I'd love to move to `extern { type Foo; }`, but IIRC it requires a `#[feature]` and I'd like to keep those limited if at all possible (and AFAIK there's no other way to create an unsized yet no wide pointer type)